platforms/scripts/afu_synth_setup_std

```diff
@@ -0,0 +1,286 @@
+    else:
+        try:
+            byte_out = subprocess.check_output(cmd)
+            str_out = byte_out.decode("utf-8")
+        except OSError as e:
+            if e.errno == os.errno.ENOENT:
+                msg = cmd[0] + " not found on PATH!"
+                errorExit(msg)
+            else:
+                raise
+        except subprocess.CalledProcessError as e:
+            sys.stderr.write(e.output)
+            raise
+
+        return str_out
+
+
+# Remove a file, if it exists
+def remove_file(f):
+    if (os.path.isfile(f) or os.path.islink(f)):
+        os.unlink(f)
+
+
+# Remove a directory tree, if it exists
+def remove_dir(d):
+    if (os.path.isdir(d)):
+        shutil.rmtree(d)
+
+
+# Copy base build environment to target directory
+def copy_build_env(base_src, dst, force):
+    # Target directory can't exist (unless force is set)
+    dst = dst.rstrip(os.path.sep)
+    if dst == '':
+        errorExit('Target directory not set')
+
+    if (os.path.exists(dst)):
+        if (os.path.islink(dst)):
+            errorExit('Target ({0}) is a link.'.format(dst))
+        if (not force):
+            errorExit('Target ({0}) already exists.'.format(dst))
+        if (os.path.isdir(dst)):
+            # Clean up inside the existing directory
+            remove_dir(os.path.join(dst, 'build'))
+            remove_dir(os.path.join(dst, 'scripts'))
+            # Drop all top-level files in the 'hw' directory
+            for f in glob.glob(os.path.join(dst, 'hw/*')):
+                remove_file(f)
+            # Drop 'hw' if it is a file
+            remove_file(os.path.join(dst, 'hw'))
+        else:
+            os.remove(dst)
+            os.mkdir(dst)
+    else:
+        os.mkdir(dst)
+
+    # Copy build to target directory
+    build_src = os.path.join(base_src, 'hw/lib/build')
+    build_dst = os.path.join(dst, 'build')
+    print('Copying build from {0}...'.format(build_src))
+    try:
+        shutil.copytree(build_src, build_dst)
+    except Exception:
+        shutil.rmtree(dst)
+        print('Failed to copy {0} to {1}'.format(build_src, build_dst))
+        raise
+
+    # Make target "hw" directory
+    hw_dir = os.path.join(dst, 'hw')
+    if (not os.path.isdir(hw_dir)):
+        os.mkdir(hw_dir)
+
+    # Make a scripts directory with clean.sh and run.sh
+    scripts_dst = os.path.join(dst, 'scripts')
+    os.mkdir(scripts_dst)
+    scripts_src = os.path.relpath(os.path.join(base_src, 'bin'), scripts_dst)
+    os.symlink(os.path.join(scripts_src, 'clean.sh'),
+               os.path.join(scripts_dst, 'clean.sh'))
+    run_sh = os.path.join(scripts_dst, 'run.sh')
+    with open(run_sh, 'w') as f:
+        # Use relative paths so generated scripts can be shipped in
+        # a tarball along with a release.
+        f.write('#!/bin/sh\n')
+        f.write('SCRIPT_DIR="$(cd "$(dirname -- "$0")" 2>/dev/null && ' +
+                'pwd -P)"\n')
+        f.write('RELEASE_DIR="$(readlink -f ${SCRIPT_DIR}/' +
```

Comment:
I'd `"`-wrap the `$SCRIPT_DIR/foo` argument passed to readlink to avoid unintended word splitting.